CI_README.md

```diff
@@ -0,0 +1,51 @@
+
+Once a set of downstream commits have been created, the Magician pushes them directly to the downstream's `master` branch. It doesn't use the `--force` flag. If the branch goes out of sync with what MM expects, the push will fail without damaging the downstream repository's history.  When a push fails, `downstream-master` will not be updated, and the commits will be reprocessed in the next sync. 
+
+We expect the failure case to be somewhat uncommon. Downstream repos are generally only edited directly to vendor new libraries or update READMEs or other non-MM-managed files.
+
+It is safe to have more than one downstream-push running at the same time due to this property, in the event of overruns.  Each run will either
+a) make no changes to any downstream and fail
+or
+b) atomically update every downstream to a fast-forward state that represents the appropriate HEAD as of the beginning of the run
+
+It's possible, if we assume the worst, for a job to be cancelled or fail in the middle of pushing downstreams, if lightning strikes a datacenter or some other unlikely misfortune happens.  This has a chance to cause a hiccup in the downstream history, but isn't dangerous.
+
+## Deploying the pipeline
+The code on the `master` branch is used in all cases - if you are making changes to the Actions, your changes will not apply until they are merged in.
+
+## Design choices & tradeoffs
+* The downstreams share some setup code in common - especially TPG and TPGB.  We violated the DRY principle by writing separate workflows for each repo.  In practice, this has substantially reduced the amount of code - the coordination layer above the two repos was larger than the code saved by combining them.  We also increase speed, since each Action runs separately.
+* The downstream push doesn't happen immediately.  We caused some delay with the cron-based approach that could have been avoided.  This ensures that two commits which are merged at about the same time will never conflict, since only one copy of the downstream push will be running at a time.
+* The downstream push doesn't open PRs against downstreams.  This may inconvenience some existing workflows which rely on the downstream PRs.  This ensures that merge conflicts never come into play, since the downstreams never have dangling PRs.
+* The downstream push is totally disconnected from the differ.  This means that the diff which is approved isn't guaranteed to be applied *exactly*, if for instance magic modules' behavior changes on master between diff generation and downstream push.  This is also intended to avoid merge conflicts by, effectively, rebasing each commit on top of master before final generation is done.
+* We deleted the submodules.  They weren't useful to us and they were annoying to update - no one ever used them for anything as far as we know - but it's possible there's some long-forgotten workflow that someone is using which will be damaged.
```

Comment:
> no one ever used them for anything as far as we know

I mean strictly speaking this isn't true, it's just that nobody is using them _anymore_